discovery_engine_core/web-api/openapi.yaml

```diff
@@ -128,12 +128,174 @@ paths:
+      description: Gets the property of the document.
+      parameters:
+        - name: document_id
+          in: path
+          description: Id of the document
+          required: true
+          schema:
+            $ref: '#/components/schemas/DocumentId'
+        - name: property_id
+          in: path
+          description: Id of the document property
+          required: true
+          schema:
+            $ref: '#/components/schemas/DocumentPropertyId'
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DocumentProperty'
+        '400':
+          description: invalid document id or property id
+        '404':
+          description: document id or property id not found
+    put:
+      tags:
+        - documents
+      summary: Set a document property
+      description: Sets or replaces the property of the document.
+      parameters:
+        - name: document_id
+          in: path
+          description: Id of the document
+          required: true
+          schema:
+            $ref: '#/components/schemas/DocumentId'
+        - name: property_id
+          in: path
+          description: Id of the document property
+          required: true
+          schema:
+            $ref: '#/components/schemas/DocumentPropertyId'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/DocumentProperty'
+      responses:
+        '204':
+          description: successful operation
+        '400':
+          description: invalid document id or property id
+        '404':
+          description: document id not found
+    delete:
+      tags:
+        - documents
+      summary: Delete a document property
+      description: Deletes the property of the document.
+      parameters:
+        - name: document_id
+          in: path
+          description: Id of the document
+          required: true
+          schema:
+            $ref: '#/components/schemas/DocumentId'
+        - name: property_id
+          in: path
+          description: Id of the document property
+          required: true
+          schema:
+            $ref: '#/components/schemas/DocumentPropertyId'
+      responses:
+        '204':
+          description: successful operation
+        '400':
+          description: invalid document id or property id
+        '404':
+          description: document id or property id not found
+
 components:
   schemas:
     DocumentId:
       description: A document id can be any non-empty UTF-8 string that does not contain the null byte.
       type: string
       example: "document id"
+    DocumentPropertyId:
+      description: A document property id can be any non-empty UTF-8 string that does not contain the null byte.
+      type: string
+      example: "document property id"
+    DocumentProperty:
```

Comment:
:grey_question: 
Shouldn't this be `DocumentProperties` since it contains all the properties of a document?

Reply:
no, this is a single property. all properties are defined in l.299.

Reply:
how is the map typed correctly? i guess `type: Map[string, DocumentProperty]` won't do it, how do we correctly declare a `$ref` to `DocumentProperty` there?

Reply:
`{}` means any type. `DocumentProperties should be a map from `DocumentPropertyId` to `DocumentProperty(Value?)`.
As done for other requests/responses we can wrap the content in an object like
```
{
  value: DocumentProperty
 }
 ```
 and something like
 ```
 {
   properties: DocumentProperties
 }
 ```

Reply:
not sure what you are trying to say, it seems to refer to the value/instance of a `DocumentProperties`, but not the `type` of one. anyways, i found the corresponding section in the docs and changed it accordingly: https://spec.openapis.org/oas/v3.1.0#model-with-map-dictionary-properties

Reply:
We should not return the DocumentProperties raw but wrapper in an object so that in the future we can add other data. The same for the request, if we wrap the value in an object will be easier to add new functionality later.
If the same reason for which, for example, `IngestionRequest` as a field `documents` instead of taking the list directly.

Reply:
ah ok, got it, i added the wrappers.